app/notification.go

```diff
@@ -544,8 +544,17 @@ func sendToPushProxy(msg model.PushNotification) *model.AppError {
 	if resp, err := httpClient.Do(request); err != nil {
 		return model.NewLocAppError("sendToPushProxy", "api.post.send_notifications_and_forget.push_notification.error", map[string]interface{}{"DeviceId": msg.DeviceId, "Error": err.Error()}, "")
 	} else {
-		ioutil.ReadAll(resp.Body)
-		resp.Body.Close()
+		m := model.MapFromJson(resp.Body)
+		if resp.Body != nil {
+			ioutil.ReadAll(resp.Body)
+			resp.Body.Close()
+		}
+
+		if m != nil && m[model.STATUS] == model.STATUS_REMOVE {
```

Comment:
Minor, but unneeded nil check since `MapFromJson` always returns a valid map. Left up to your discretion as to whether or not to remove